arraycontext/impl/pytato/__init__.py

```diff
@@ -322,6 +359,79 @@ def __init__(
 
         self._force_svm_arg_limit = _force_svm_arg_limit
 
+    def _wait_and_transfer_profile_events(self) -> None:
+        # First, wait for completion of all events
+        if self.profile_events:
+            cl.wait_for_events([p_event.cl_event for p_event in self.profile_events])
+
+        # Then, collect all events and store them
+        for t in self.profile_events:
+            name = t.translation_unit.program.entrypoint
+
+            time = t.cl_event.profile.end - t.cl_event.profile.start
+
+            self.profile_results.setdefault(name, []).append(time)
+
+        self.profile_events = []
+
+    def get_profiling_data_for_kernel(self, kernel_name: str) \
+          -> MultiCallKernelProfile:
+        """Return profiling data for kernel `kernel_name`."""
+        self._wait_and_transfer_profile_events()
+
+        time = 0
+        num_calls = 0
+
+        if kernel_name in self.profile_results:
+            knl_results = self.profile_results[kernel_name]
+
+            num_calls = len(knl_results)
+
+            for r in knl_results:
+                time += r
+
+        return MultiCallKernelProfile(num_calls, time)
+
+    def reset_profiling_data_for_kernel(self, kernel_name: str) -> None:
+        """Reset profiling data for kernel `kernel_name`."""
+        self.profile_results.pop(kernel_name, None)
+
+    def tabulate_profiling_data(self) -> pytools.Table:
```

Comment:
This looks more like a helper function than a class method?

Reply:
Dunno - this is currently the main way of "getting" the complete profiling data.

Reply:
Isn't the main way of getting it `actx.profile_results` and/or `actx.get_profile_for_kernel`? This is just a handy way to display it (probably one of many).

Reply:
I agree with @alexfikl that this should be a function, not a method.

Reply:
What do you guys think of [1123c16](https://github.com/inducer/arraycontext/pull/311/commits/1123c16c25fd9b6c114fd1c2031cef963fedcf4b)?

Reply:
Looks good to me!